packages/ui/src/types/accountSettings/validator.ts

```diff
@@ -0,0 +1,9 @@
+export type InputEventType = 'blur' | 'change';
+
+export type ValidationMode = 'onBlur' | 'onChange' | 'onTouched';
+
+export interface FieldValidator {
+  validationMode: ValidationMode;
+  validate: (field: string) => boolean;
```

Comment:
Agree with suggested change, done in https://github.com/aws-amplify/amplify-ui/pull/2900/commits/ff5dc1081cc18e92350ccca7546c16c4b00f4210.

Reply:
I do not think we need to accept async validators yet, because these are field validators that run on every change/blur. 

I believe async validator typically happens in `onSubmit`, which we do not support yet. Wwe can add async support as an additive feature when we support `onSubmit` validators.